samcli/lib/test_runner/test_runner_template_generator.py

```diff
@@ -0,0 +1,347 @@
+    image_uri: str,
+    vpc_id: str,
+    cpu: int,
+    memory: int,
+) -> dict:
+    """
+    Renders a base jinja template with a set of parameters needed to create the Test Runner Stack.
+
+    Parameters
+    ----------
+    jinja_template_json_string : str
+        The base jinja template to which parameters are substituted.
+
+    bucket_name : str
+        The name of the S3 bucket used by the Test Runner Stack.
+
+    ecs_task_exec_role_arn : str
+        The ARN of the AWS ECS Task Execution Role used by the Test Runner Stack.
+
+        NOTE: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_execution_IAM_role.html
+
+    image_uri : str
+        The URI of the Image to be used by the Test Runner Fargate task definition.
+
+    vpc_id : str
+        The ID of the VPC associated with the Test Runner security group.
+
+        NOTE: The security group (and associated VPC) is required to invoke the runTask API and run tests
+
+    cpu : int
+        The CPU used by the Test Runner Task Definition.
+
+        NOTE: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html#task_size
+
+    memory : int
+        The memory used by the Test Runner Task Definition.
+
+        NOTE: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html#task_sizes
+
+    Returns
+    -------
+    dict
+        A JSON object representing the CloudFormation Template for the Test Runner Stack with all parameters substituted in.
+
+    """
+    data = {
+        "cpu": cpu,
+        "memory": memory,
+        "vpc_id": vpc_id,
+        "ecs_task_exec_role_arn": ecs_task_exec_role_arn,
+        "image_uri": image_uri,
+        "s3_bucket_name": bucket_name,
+    }
+
+    try:
+        rendered_template_string = Template(jinja_template_json_string).render(data)
+        return json.loads(rendered_template_string)
+    except Exception as render_error:
+        LOG.exception("Failed to render jinja template: %s", render_error)
+        return None
+
+
+def __query_tagging_api(tag_filters: dict) -> dict:
+    """
+    Queries the Tagging API to retrieve the ARNs of every resource with the given tags.
+
+    NOTE: https://docs.aws.amazon.com/cli/latest/reference/resourcegroupstaggingapi/get-resources.html#get-resources
+
+    Parameters
+    ----------
+    tag_filters : dict
+        The tag filters to restrict output to only those resources with the given tags.
+
+        NOTE: https://docs.aws.amazon.com/cli/latest/reference/resourcegroupstaggingapi/get-resources.html#options
+
+    Returns
+    -------
+    dict
+        A response object containing ResourceTagMappingList, which contains a list of resource ARNs and tags associated with each.
+
+        NOTE: https://docs.aws.amazon.com/cli/latest/reference/resourcegroupstaggingapi/get-resources.html#output
+
+
+    """
+    try:
+        return boto3.client("resourcegroupstaggingapi").get_resources(tag_filters=tag_filters)
```

Comment:
If only `ResourceTagMappingList` is accessed anyway, might consider returning that directly.

Reply:
Mhmm, can easily refactor back if we need other data from the response object